src/main/java/com/gluonhq/substrate/util/FileOps.java

```diff
@@ -307,4 +323,48 @@ public static String calculateCheckSum(File file) {
             return "";
         }
     }
+
+    /**
+     * Extracts the files that match a given extension found in a jar to a target patch,
+     * providing that the file passes a given filter, and it doesn't exist yet in the target path
+     *
+     * @param extension the extension of the files in the jar that will be extracted
+     * @param sourceJar the path to the jar that will be inspected
+     * @param target the path of the folder where the files will be extracted
+     * @param filter a predicate that the files in the jar should match.
+     * @throws IOException
+     */
+    public static void extractFilesFromJar(String extension, Path sourceJar, Path target, Predicate<Path> filter) throws IOException {
+        if  (!Files.exists(sourceJar)) {
+            return;
+        }
+        if  (!Files.exists(target)) {
+            Files.createDirectories(target);
```

Comment:
What if target already exists? How can that happen? In that case, is the processing still relevant?

Reply:
target is the folder where the files will be extracted to. It could exist from a previous run, or if we call several times this function (i.e. from different jars). We only need to create it the first time.

Reply:
But if it exists and it contains wrong data (e.g. from previous runs), wouldn't that lead to issues with wrong symbols? Wouldn't it be better to remove the directory at the beginning of the cycle, create the directory on the first occasion it is needed, and bail if it already exists?